### PR TITLE
[js] Added 'ConfigOptionProvider' to provide filenames ending with '.js' in Profile configuration

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added 'ConfigOptionProvider' to provide filenames ending with '.js' in Profile configuration

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.transform.javascript/src/main/java/org/openhab/transform/javascript/internal/profiles/JavaScriptTransformationProfile.java [hunk 1]
```diff
@@ -52,7 +51,7 @@
     @NonNullByDefault({})
```
Comment (J-N-K) [inline bundles/org.openhab.transform.javascript/src/main/java/org/openhab/transform/javascript/internal/profiles/JavaScriptTransformationProfile.java:51]: This annotation is wrong. l. 75 sets it to `null`. (also above)

**Review by J-N-K — CHANGES_REQUESTED** [review]
otherwise looks good

File: bundles/org.openhab.transform.javascript/src/main/java/org/openhab/transform/javascript/internal/profiles/JavaScriptTransformationProfile.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.transform.javascript/src/main/java/org/openhab/transform/javascript/internal/profiles/JavaScriptTransformationProfile.java]: Absolutely. I changed it.

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
Can you resolve conflicts?

**Comment by cweitkamp** [comment]
Rebase done.

**Pull request merged**